<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>WalletRpcSettingsDialog</class>
 <widget class="QDialog" name="WalletRpcSettingsDialog">
  <property name="windowModality">
   <enum>Qt::ApplicationModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>588</width>
    <height>358</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Wallet RPC settings</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QCheckBox" name="m_enableWalletRpcCheck">
     <property name="text">
      <string>Run Wallet RPC interface</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>Interface</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <property name="spacing">
         <number>10</number>
        </property>
        <property name="sizeConstraint">
         <enum>QLayout::SetMaximumSize</enum>
        </property>
        <item>
         <widget class="QLabel" name="label_1">
          <property name="text">
           <string>Bind to IP:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="m_walletRpcIp">
          <property name="text">
           <string>127.0.0.1</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Port: </string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="m_walletRpcPort">
          <property name="minimum">
           <number>0</number>
          </property>
          <property name="maximum">
           <number>65535</number>
          </property>
          <property name="value">
           <number>15000</number>
          </property>
          <property name="displayIntegerBase">
           <number>10</number>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>Authorization</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <property name="sizeConstraint">
         <enum>QLayout::SetMaximumSize</enum>
        </property>
        <item>
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Username:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="m_walletRpcUser"/>
        </item>
        <item>
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>Password:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="m_walletRpcPassword">
          <property name="inputMethodHints">
           <set>Qt::ImhHiddenText|Qt::ImhNoAutoUppercase|Qt::ImhNoPredictiveText|Qt::ImhSensitiveData</set>
          </property>
          <property name="echoMode">
           <enum>QLineEdit::Password</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="m_showPwdButton">
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="../../resources.qrc">
            <normaloff>:/icons/tracking</normaloff>:/icons/tracking</iconset>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_4">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="label">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="text">
      <string>Note! Use this only in a secure environment as the password is stored in plain text config.</string>
     </property>
     <property name="textFormat">
      <enum>Qt::AutoText</enum>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>131</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="walletRpcSetetingsCancelButton">
       <property name="text">
        <string>Cancel</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="walletRpcSetetingsSaveButton">
       <property name="text">
        <string>Save</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../../resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>walletRpcSetetingsCancelButton</sender>
   <signal>clicked()</signal>
   <receiver>WalletRpcSettingsDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>626</x>
     <y>381</y>
    </hint>
    <hint type="destinationlabel">
     <x>381</x>
     <y>201</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>walletRpcSetetingsSaveButton</sender>
   <signal>clicked()</signal>
   <receiver>WalletRpcSettingsDialog</receiver>
   <slot>saveSettingClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>712</x>
     <y>381</y>
    </hint>
    <hint type="destinationlabel">
     <x>381</x>
     <y>201</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_showPwdButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>WalletRpcSettingsDialog</receiver>
   <slot>showPwdToggled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>420</x>
     <y>156</y>
    </hint>
    <hint type="destinationlabel">
     <x>293</x>
     <y>178</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>saveSettingClicked()</slot>
  <slot>showPwdToggled(bool)</slot>
 </slots>
</ui>
